src/templates/templates_config.ts

```diff
@@ -11,11 +11,31 @@ interface TemplateConfig {
   };
 }
 
+const templateRegex = /^[a-zA-Z0-9!@#$%^&*()-+=_{}|:;"'<>,.?~`]{1,64}$/ //TODO: decide on the Regex
```

Comment:
Where is this regex being applied? Is this checking against the template name, the job name, or the contents of the template itself? 

If it's for content, I think we need a pretty permissive rule since special characters are needed to define components and variables. For this case, we may want to drop the {1,64}$ at the end. I think that means the expression only checks the first 64 characters and there will be more characters than that in a template.

If the expression is for either the template name or project name, we should probably be pretty restrictive and keep names alphanumeric with spaces allowed

Reply:
THis is for template name (such as doe_workflow_hpwh), which will be appare in a link, so I don't think space should be allowed